### ISSUE TITLE
language defaults to japanese currently on emacswiki?

### ISSUE THREAD

**Issue by steckerhalter** [emacsmirror/emacswiki.org#5]
We have been wondering about this in https://github.com/quelpa/quelpa/issues/77

Now it happened again and it seems that the default language isn't english anymore but japanese (see for example http://www.emacswiki.org/emacs/open-junk-file.el)

I'm afraid but `package-build.el` (from melpa) is trying to get the time stamp of the last edit via `Last edited ...` and that fails of course if it is in japanese.

Do you know why it happens?

**Comment by wasamasa** [comment]
Ping @nicferrier @kensanata

**Comment by kensanata** [comment]
This is a bug in the cache setup that we can't seem to fix. Requests are handled by nginx and served from the cache if possible. We've been unable to set it up such that the correct language is being served. This means that essentially visitors will get a random language. If you have a cookie set, however, or if you use a magic bypass URL parameter, you will bypass the nginx cache and be served by the original Apache server instead. Apache and the wiki handle your requests correctly. There's an HTTP header that will tell you whether you got served a cache hit or not.

Example 1:
```
alex@Megabombus:~$ curl -I http://www.emacswiki.org/emacs/open-junk-file.el
HTTP/1.1 200 OK
Server: nginx/1.4.6 (Ubuntu)
Date: Thu, 20 Aug 2015 04:33:05 GMT
Content-Type: text/html; charset=utf-8
Connection: keep-alive
Vary: Accept-Language
X-EmacsWiki-Source: Apache/CGI
X-Proxy-Cache: HIT
```

Example 2:
```
alex@Megabombus:~$ curl -I "http://www.emacswiki.org/emacs/open-junk-file.el?bypass=1"
HTTP/1.1 200 OK
Server: nginx/1.4.6 (Ubuntu)
Date: Thu, 20 Aug 2015 04:35:00 GMT
Content-Type: text/html; charset=utf-8
Connection: keep-alive
Vary: Accept-Language
X-EmacsWiki-Source: Apache/CGI
X-Proxy-Cache: BYPASS
```

**Comment by kensanata** [comment]
You might be better off scraping less HTML and using markup instead of language strings.
```
alex@Megabombus:~$ curl --silent http://www.emacswiki.org/emacs/AlexSchroeder | xmllint --html --xpath '//div[@class="time"]' -| sed -e 's/.*\([0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9] [0-9][0-9]:[0-9][0-9]\).*/\1/'
2015-04-09 18:51
```
Unfortunately, it seems that some pages don't contain valid HTML.

**Comment by kensanata** [comment]
Or, if you want to keep scraping, you might force the interface lange.
```
alex@Megabombus:~$ curl --silent "http://www.emacswiki.org/emacs/open-junk-file.el?interface=zh-cn;cache=0" | xmllint --html --xpath '//div[@class="time"]' -
<div class="time">最后编辑于 2013-01-31 07:20 UTC 由 <a href="http://www.emacswiki.org/emacs/rubikitch" class="author" title="61-26-108-88.rev.home.ne.jp">rubikitch</a> <a rel="nofollow" href="http://www.emacswiki.org/emacs?action=browse;diff=2;id=open-junk-file.el" class="diff">(比较差异)</a></div>
```

**Comment by steckerhalter** [comment]
Ok, thanks for the info. I think forcing the language would be the best for now. How about if we would see to get the source code from this repo here instead of the wiki itself. Would that be a better solution for the future?

**Comment by steckerhalter** [comment]
english would be:

```
http://www.emacswiki.org/emacs/open-junk-file.el?interface=en;cache=0
```

?

**Comment by kensanata** [comment]
> I think forcing the language would be the best for now.
I don't agree. There should be some minimal HTML parsing in order to extract the right div. If you don't, how you prevent any ol' "Last edited 2013-01-31 07:20" from confusing you? Any change you make will make your code more brittle.
> english would be:
> http://www.emacswiki.org/emacs/open-junk-file.el?interface=en;cache=0
Yes

Note that you can also get the source code of wiki pages, ie.
```
http://www.emacswiki.org/emacs/raw/open-junk-file.el
```
And the wiki is smart enough to return the correct mtime in the `Last-Modified` header:
```
alex@Megabombus:~$ curl -I http://www.emacswiki.org/emacs/raw/open-junk-file.el
HTTP/1.1 200 OK
Server: nginx/1.4.6 (Ubuntu)
Date: Thu, 20 Aug 2015 08:36:56 GMT
Content-Type: text/plain; charset=utf-8
Connection: keep-alive
Vary: Accept-Language
Last-Modified: Thu, 31 Jan 2013 07:20:55 GMT
X-EmacsWiki-Source: Apache/CGI
X-Proxy-Cache: HIT
```

**Comment by steckerhalter** [comment]
ok that sounds like a better approach... to get the date from the header

**Comment by steckerhalter** [comment]
So for our use-case the header should be good enough.

It would be still good if the error with the language in the wiki could be fixed sometimes so I'm leaving the issue open
